projects/observability/src/pages/explorer/explorer-dashboard-builder.ts

```diff
@@ -56,7 +57,261 @@ export class ExplorerDashboardBuilder {
+            'col-start': 2,
+            'col-end': 3,
+            'row-start': 1,
+            'row-end': 2
+          },
+          {
+            // First row
+            type: 'cell-span-model',
+            'col-start': 3,
+            'col-end': 4,
+            'row-start': 1,
+            'row-end': 2
+          }
+        ],
+        'enable-style': true,
+        'grid-gap': '0'
+      };
+    }
+
+    return {
+      type: 'custom-container-layout',
+      'column-dimensions': [
+        {
+          type: 'dimension-model',
+          dimension: 1,
+          unit: 'FR'
+        },
+        {
+          type: 'dimension-model',
+          dimension: 1,
+          unit: 'FR'
+        },
+        {
+          type: 'dimension-model',
+          dimension: 1,
+          unit: 'FR'
+        },
+        {
+          type: 'dimension-model',
+          dimension: 1,
+          unit: 'FR'
+        }
+      ],
+      'row-dimensions': [
+        {
+          type: 'dimension-model',
+          dimension: 124,
+          unit: 'PX'
+        },
+        {
+          type: 'dimension-model',
+          dimension: 124,
+          unit: 'PX'
+        }
+      ],
+      'cell-spans': [
+        {
+          // First row
+          type: 'cell-span-model',
+          'col-start': 0,
+          'col-end': 1,
+          'row-start': 0,
+          'row-end': 1
+        },
+        {
+          type: 'cell-span-model',
+          'col-start': 1,
+          'col-end': 2,
+          'row-start': 0,
+          'row-end': 1
+        },
+        {
+          type: 'cell-span-model',
+          'col-start': 2,
+          'col-end': 3,
+          'row-start': 0,
+          'row-end': 1
+        },
+        {
+          type: 'cell-span-model',
+          'col-start': 3,
+          'col-end': 4,
+          'row-start': 0,
+          'row-end': 1
+        },
+        {
+          // Second row
+          type: 'cell-span-model',
+          'col-start': 0,
+          'col-end': 1,
+          'row-start': 1,
+          'row-end': 2
+        },
+        {
+          type: 'cell-span-model',
+          'col-start': 1,
+          'col-end': 2,
+          'row-start': 1,
+          'row-end': 2
+        },
+        {
+          type: 'cell-span-model',
+          'col-start': 2,
+          'col-end': 3,
+          'row-start': 1,
+          'row-end': 2
+        },
+        {
+          type: 'cell-span-model',
+          'col-start': 3,
+          'col-end': 4,
+          'row-start': 1,
+          'row-end': 2
+        }
+      ],
+      'enable-style': true,
+      'grid-gap': '0'
+    };
+  }
+
   private buildVisualizationDashboard(request: ExploreVisualizationRequest): Observable<ExplorerGeneratedDashboard> {
+    if (request.interval === undefined && request.groupBy === undefined) {
+      return of({
+        json: {
+          type: 'container-widget',
+          layout: this.getLayoutForElements(request.series?.length),
```

Comment:
Do we need the `?` here, if we do, then do we need `request.series?.length ?? 0`?
The method `getLayoutForElements` is expecting a number, so, unless `series` is always there, it might break.

Reply:
Only needed to add the `?` because of [this test][1]. Tried adding some data in the mock request but it doesn't pass through, yelling `series` is undefined. In the app, `series` is always defined since UI always has at least 1 metric.

[1]: https://github.com/razorpay/hypertrace-ui/pull/107/files#diff-27333c93103a04c2b6019f391fc9addc56af21234424398dc7b92516147e2685R36